File: Data Structures & Algorithms/longest-consecutive-sequence/submission-6.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    longestConsecutive(nums) {
        const checker = new Set(nums);
        let res = 0
        for (let i = 0; i < nums.length; i++){
            if (!(checker.has(nums[i] - 1))){
                let curr = 0
                while (checker.has(nums[i] + curr)){
                    curr++;
                }
                res = Math.max(res, curr)
            }
        }
        return res;
    }
}

File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-6.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        l = 0
        res = 0 
        frequency = {}
        maxF = 0
        for r in range(len(s)):
            frequency[s[r]] = frequency.get(s[r], 0) + 1
            maxF = max(maxF, frequency[s[r]])

            while r - l + 1 - maxF > k: 
                frequency[s[l]] -= 1
                l += 1
            
            res = max(res, r - l + 1)
        
        return res 
            